File: java-practice/spring-security-6-multi-filter/src/main/java/org/multifilter/security/BaseSecurity.java
package org.multifilter.security;


import lombok.RequiredArgsConstructor;
import org.multifilter.security.handler.BaseAccessDeniedHandler;
import org.multifilter.security.handler.BaseAuthenticationEntryPoint;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
import org.springframework.security.config.annotation.web.configurers.HeadersConfigurer;
import org.springframework.stereotype.Component;

import static org.springframework.security.config.http.SessionCreationPolicy.STATELESS;

@Component
@RequiredArgsConstructor
public class BaseSecurity extends AbstractHttpConfigurer<BaseSecurity, HttpSecurity> {
    private final BaseAccessDeniedHandler customAccessDeniedHandler;
    private final BaseAuthenticationEntryPoint customAuthenticationEntryPoint;

    private boolean flag;

    @Override
    public void init(HttpSecurity http) throws Exception {
        if (flag) {
            http
                    .csrf(AbstractHttpConfigurer::disable)
                    .cors(AbstractHttpConfigurer::disable)
                    .httpBasic(AbstractHttpConfigurer::disable)

                    .headers(h -> h.frameOptions(HeadersConfigurer.FrameOptionsConfig::disable))
                    .sessionManagement(s -> s.sessionCreationPolicy(STATELESS))

                    .exceptionHandling(exceptionHandlingCustomizer -> exceptionHandlingCustomizer
                            .accessDeniedHandler(customAccessDeniedHandler)
                            .authenticationEntryPoint(customAuthenticationEntryPoint))
            ;
        }
    }

    public void active() {
        this.flag = true;
    }
}

File: server-practice/message-server/security/security-webflux/src/main/kotlin/dev/lkdcode/security/filter/base/BaseSecurity.kt
package dev.lkdcode.security.filter.base

import dev.lkdcode.security.handler.ApiAccessDeniedHandler
import dev.lkdcode.security.handler.ApiAuthenticationEntryPoint
import org.springframework.security.config.web.server.ServerHttpSecurity
import org.springframework.security.web.server.context.NoOpServerSecurityContextRepository
import org.springframework.stereotype.Component

@Component
class BaseSecurity(
    private val baseAuthenticationHandler: ApiAuthenticationEntryPoint,
    private val baseDeniedHandler: ApiAccessDeniedHandler,
    private val corsSecurity: CorsSecurity,
) {

    fun init(http: ServerHttpSecurity) {
        http.exceptionHandling {
            it.authenticationEntryPoint(baseAuthenticationHandler)
                .accessDeniedHandler(baseDeniedHandler)
        }
            .csrf(ServerHttpSecurity.CsrfSpec::disable)
            .httpBasic(ServerHttpSecurity.HttpBasicSpec::disable)
            .formLogin(ServerHttpSecurity.FormLoginSpec::disable)
            .headers { it.frameOptions { option -> option.disable() } }
            .securityContextRepository(NoOpServerSecurityContextRepository.getInstance())
            .cors { it.configurationSource(corsSecurity.corsConfigurationSource()) }
    }
}